libs/core/src/components/sage-tooltip/sage-tooltip.tsx

```diff
@@ -0,0 +1,205 @@
+    }
+  }
+
+  /**
+   * Determines whether or not the tooltip have an arrow
+   */
+  @Prop({mutable: true}) disabled = false;
+
+  // eslint-disable-next-line @stencil/no-unused-watch
+  @Watch('disabled')
+  handleDisabled() {
+    // console.log('disabled')
+  }
+
+  /**
+   * Emitted after a tooltip is closed
+   */
+  @Event() sageTooltipHide: EventEmitter;
+
+  /**
+   * Emitted after a tooltip is shown
+   */
+  @Event() sageTooltipShow: EventEmitter;
+
+  componentWillLoad() {
+    if (this.opened) {
+      this.showTooltip();
+    }
+
+    this.el.addEventListener('blur', this.handleHide, true);
+    this.el.addEventListener('focus', this.handleShow, true);
+  }
+
+  componentDidUpdate() {
+    if (this.opened) {
+      this.showTooltip();
+    }
+  }
+
+  componentDidRender() {
+    positionTooltip(this.el, this.placement, this.contentEl);
+  }
+
+  @Method()
+  async showTooltip() {
+    if(!this.disabled) {
+      this.opened = true;
+
+      if(this.contentEl) {
+        // TODO: need to use block / none but the tooltip content width and height are needed for calculations
+        // this.contentEl.style.display = 'block';
+        this.contentEl.style.opacity = '1';
+        this.contentEl.style.visibility = 'visible';
+      }
+    }
+  }
+
+  @Method()
+  async hideTooltip() {
+    this.opened = false;
+    // TODO: need to use block / none but the tooltip content width and height are needed for calculations
+    // this.contentEl.style.display = '';
+    this.contentEl.style.opacity = '0';
+    this.contentEl.style.visibility = 'hidden';
+  }
+
+  private handleHide = () => {
+    this.hideTooltip();
+    this.sageTooltipHide.emit();
+  };
+
+  private handleShow = () => {
+    if(this.disabled) {
+      return
+    }
+    this.showTooltip();
+    this.sageTooltipShow.emit();
+  };
+
+  render() {
+    return (
+      <Host
+        class={{'sage-tooltip--has-html-content': this.htmlContent}}
+        disabled={this.disabled}
+        hasArrow={this.hasArrow}
+        onMouseEnter={this.handleShow}
+        onMouseLeave={this.handleHide}
+        onFocusin={this.handleShow}
+        onFocusout={this.handleHide}
+      >
+        <div
+          class={`
+            sage-tooltip
+            ${this.placement ? (`sage-tooltip--${this.placement}`) : ''}
+            ${this.opened ? 'sage-tooltip--is-open' : ''}
+          `}
+        >
+          <span
+            aria-describedby={this.componentId}
+            class="sage-tooltip__trigger"
+          >
+            <slot />
+          </span>
+
+          <div class="sage-tooltip__content"
+            aria-hidden={this.opened ? 'false' : 'true'}
+            aria-live={this.opened ? 'polite' : 'off'}
+            id={this.componentId}
+            ref={(el) => (this.contentEl = el)}
+            role="tooltip"
+          >
+            <slot
+              name="content"
+            ></slot>
+            {this.content}
+            {this.hasArrow && (
+              <div
```

Comment:
Any thoughts on possibly using a class and pseudo-element (`:after`) here to remove the need for an empty div?

Reply:
Not currently. What's wrong with having that empty div?

Reply:
I'm with Phillip here, if we can build the arrow without adding a div to the DOM I'm for it. It's purely presentational so I'd lean toward a pseudo-element too.

Reply:
Updated to use pseudo element and update tests